pkg/tsdb/postgres/postgres.go

```diff
@@ -18,15 +21,19 @@ import (
 
 func init() {
 	tsdb.RegisterTsdbQueryEndpoint("postgres", newPostgresQueryEndpoint)
+
 }
 
-func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
-	logger := log.New("tsdb.postgres")
-	logger.Debug("Creating Postgres query endpoint")
+func cleanUpFiles(config sslAuthenticationConfig) {
+	if config.tmpFilesPath != "" {
+		defer os.RemoveAll(config.tmpFilesPath)
```

Comment:
Lint complains because you should check the error returned value from `os.RemoveAll`.

Also, I think that you don't need `defer` here. Defer is used when we want to execute something after the surrounding function returns. So probably it's required [when cleanUpFiles is called](https://github.com/grafana/grafana/blob/58ade53a2b62245f41243b356e38275dc4cffcca/pkg/tsdb/postgres/postgres.go#L70).

Reply:
Agree with @papagian that it needs to be ensured that the temporary directory gets removed. Regarding error check, you probably just want to log a warning if the temporary directory fails to be removed.

Reply:
actually, I wanted to ask you guys if I put the directory name which is dynamic, there would be risks if the program crashed in the middle, there would be "trash files" forever. how about make it permanent? would it be a concurrent kind of problem?

Reply:
It would require some more thought, but I don't think it should be a problem to create a temporary directory and ensure it gets cleaned up.